annotations/src/main/java/com/javadiscord/jdi/core/processor/ClassFileUtil.java

```diff
@@ -109,7 +110,7 @@ private static File extractClassFileFromJar(
         ZipInputStream zip,
         String entryName
     ) throws IOException {
-        File tempFile = File.createTempFile(entryName.replace('/', '_'), ".class");
+        File tempFile = Files.createTempFile(entryName.replace('/', '_'), ".class").toFile();
```

Comment:
Great spot, to make this compliant with Sonar, please could the compliant suggestion be made:

`File.createTempFile("prefix", "suffix", new File("/mySecureDirectory"));  // Compliant`

This uses the additional `File` parameter to a secure directory.